tests/integration_tests.rs

```diff
@@ -759,6 +759,136 @@ fn simple_for_dictmerger_loop() {
     unsafe { weld_value_free(ret_value) };
 }
 
+fn simple_groupmerger() {
+    #[derive(Clone)]
+    #[allow(dead_code)]
+    struct VecPair {
+        ele1: i32,
+        ele2: WeldVec<i32>,
+    }
+
+    #[allow(dead_code)]
+    struct Args {
+        x: WeldVec<i32>,
+        y: WeldVec<i32>,
+    }
+
+    let code = "|x:vec[i32], y:vec[i32]| tovec(result(for(zip(x,y), groupmerger[i32,i32], \
+                |b,i,e| merge(b, e))))";
+
+    let conf = default_conf();
+    let keys = [1, 2, 2, 1, 3, 3];
+    let vals = [2, 3, 4, 2, 1, 0];
+    let ref input_data = Args {
+        x: WeldVec {
+            data: &keys as *const i32,
+            len: keys.len() as i64,
+        },
+        y: WeldVec {
+            data: &vals as *const i32,
+            len: vals.len() as i64,
+        },
+    };
+
+    let ret_value = compile_and_run(code, conf, input_data);
+    let data = unsafe { weld_value_data(ret_value) as *const WeldVec<VecPair> };
+    let result = unsafe { (*data).clone() };
+
+    let output_keys = [1, 2, 3];
+    let output_vals = [[2,2], [3,4], [1, 0]];
+
+    assert_eq!(result.len, output_keys.len() as i64);
+    for i in 0..(output_keys.len() as isize) {
+        let mut success = false;
+        let key = unsafe { (*result.data.offset(i)).ele1 };
+        let value_vec = unsafe { ((*result.data.offset(i)).ele2).clone() };
+        for j in 0..(output_keys.len()) {
+            if output_keys[j] == key {
+                assert_eq!(value_vec.len, output_vals[j].len() as i64);
+                success = true;
+                for k in 0..(output_vals[j].len()) {
+                    let value = unsafe { *value_vec.data.offset(k as isize) };
+                    assert_eq!(output_vals[j][k], value)
+                }
+            }
+        }
+        assert_eq!(success, true);
+    }
```

Comment:
This seems like a pretty awkward way to compare two maps; can we do something else, like extract all the key-value pairs and sort them? Or create some helper functions for this?